src/Adapter/Tax/TaxComputer.php

```diff
@@ -100,15 +87,11 @@ public function computePriceWithoutTaxes(
      */
     public function getTaxRate(TaxRulesGroupId $taxRulesGroupId, CountryId $countryId): DecimalNumber
     {
-        $taxRulesGroup = $this->taxRulesGroupRepository->getTaxRulesGroupDetails($taxRulesGroupId);
-        if (!empty($taxRulesGroup['rates'])) {
-            // Use the tax rate associated to context country, or the first one as fallback
-            $countryTaxRate = $taxRulesGroup['rates'][$countryId->getValue()] ?? reset($taxRulesGroup['rates']);
-        } else {
-            $countryTaxRate = 0;
-        }
+        $address = new Address();
+        $address->id_country = $countryId->getValue();
+        $taxCalculator = TaxManagerFactory::getManager($address, $taxRulesGroupId->getValue())->getTaxCalculator();
 
-        return new DecimalNumber((string) $countryTaxRate);
+        return new DecimalNumber((string) $taxCalculator->getTotalRate());
```

Comment:
Old logic would just add last tax, now it calculates correct tax rates.
But because we are using TaxManagerFactory to do so i had to remove unit test because i couldn't mock sql that its inside getTaxCalulator() function. 
Open to suggestions on how to return and modify unit test to make it work. Putting calculator in constructor feels like the function doesn't make sense anymore.